tests/cephadm/test_restart.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from ceph.ceph_admin import CephAdmin
+
+
+log = logging.getLogger(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+      Execute the command ceph orch restart  <args>.
+
+      Args:
+          config: command and service are passed from the test case.
+
+      Example:
+          Testing ceph orch restart mon
+
+    config:
+      command: restart
+      service: mon
+
+
+      Returns:
+          output, error   returned by the command.
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    arg = config["command"]
+    ser = config["service"]
+    cmd = f"ceph orch {arg} {ser}"
+    out, err = cephadm.shell([cmd])
```

Comment:
This is a good start however it moves away from the current implementation. The current behavior has a 1 to 1 mapping with the CLI command. Also, this would result in multiple test_<start|stop|reconfig> modules... higher maintenance cost.

IMHO, we should move the behavior as a mixin that could be inherited by `orch` module. This would allow services to customize further if required when a service operation is being performed.

Performing such operations would hamper re-usability when another module would like to perform restart/stop/reapply. Moving to the lib could avoid this issue.